src/malli/json_schema.cljc

```diff
@@ -7,12 +7,21 @@
 (defprotocol JsonSchema
   (-accept [this children options] "transforms schema to JSON Schema"))
 
-(defn -ref [x] {:$ref (str "#/definitions/" x)})
+(defn -ref [x] {:$ref (apply str "#/definitions/"
+                             (cond
+                               (qualified-keyword? x) [(namespace x) "~1"
+                                                       (name x)]
```

Comment:
your  commit message on the `~1` encoding was good. perhaps a simple code comment here about that?